### ISSUE TITLE
No CPU and memory graphs on docker 1.12

### ISSUE THREAD

**Issue by netandreus** [rancher/rancher#5601]
**Rancher Version:** 1.1.2

**Docker Version:** 1.12

**OS and where are the hosts located? (cloud, bare metal, etc):** vmware esxi, ubuntu 16.04

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)** single node

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)** Cattle

**Steps to Reproduce:** Install on fresh ubuntu 16.04 on docker 1.12 from official docker repos

**Results:** empty graphs of cpu and memory for container in web-ui

**Expected:** normal graphs

**Comment by frekele** [comment]
You need to add:

'cgroup_enable' in the kernel boot to work.

adding (cgroup_enable=memory swapaccount=1) in /etc/default/grub 

````
GRUB_CMDLINE_LINUX_DEFAULT="init=/bin/systemd console=hvc0 console=ttyS0 cgroup_enable=memory swapaccount=1"
```
run update-grub and restart.

https://github.com/docker/docker/issues/18420

**Comment by frekele** [comment]
You need to add 'cgroup_enable' in the kernel boot to work.


adding (cgroup_enable=memory swapaccount=1) in /etc/default/grub 

````
GRUB_CMDLINE_LINUX_DEFAULT="init=/bin/systemd console=hvc0 console=ttyS0 cgroup_enable=memory swapaccount=1"
```
run update-grub and restart.

https://github.com/docker/docker/issues/18420

**Comment by netandreus** [comment]
Thanks a lot! It helps me!

**Comment by netandreus** [comment]
It's need to restart system twice.

**Comment by netandreus** [comment]
After some time graphs are empty again. docker stats from console reports all info.

**Comment by deniseschannon** [comment]
I don't have any issues with v1.2.0-pre2-rc3.  Are they just empty graphs?

![screen shot 2016-08-17 at 11 23 49 am](https://cloud.githubusercontent.com/assets/11410997/17748208/1f210024-646d-11e6-9ecd-ebb761279cf4.png)

**Comment by netandreus** [comment]
@deniseschannon yes, it's empty 
![rancher-empty-graphs](https://cloud.githubusercontent.com/assets/313477/17801066/7a553c10-65f0-11e6-8a69-a7b4b82eb5e3.png)

My docker version: 
Docker version 1.12.0, build 8eab29e

**Comment by frekele** [comment]
Try with docker 1.12.1
https://github.com/docker/docker/releases/tag/v1.12.1

**Comment by StrongMonkey** [comment]
@deniseschannon I believe this is already fixed in 1.2. Can we close it?

**Comment by deniseschannon** [comment]
Yes, with 1.2, we've moved away from cadvisor and are using `docker stats`. If there are no stats coming form the container, then docker isn't providing any details on it.
